supabase/functions/alert-data-processing/index.ts

```diff
@@ -0,0 +1,186 @@
+            fetch("https://api.resend.com/emails", {
+                method: "POST",
+                headers: {
+                    ...corsHeaders,
+                    "Content-Type": "application/json",
+                    "Authorization": `Bearer ${RESEND_API_KEY}`,
+                },
+                body: JSON.stringify({
+                    from: "Estuary <onboarding@resend.dev>",
+                    to: emails,
+                    subject,
+                    html: `
+                      <div style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;">
+                        ${content}
+
+                        <p style="margin-bottom: 0;">Thanks,</p>
+                        <p style="margin-top: 0;">Estuary Team</p>
+                      </div>
+                    `,
+                }),
+            }),
+    );
+
+    await Promise.all(notificationPromises);
+};
+
+serve(async (_request: Request): Promise<Response> => {
+    const startTimestamp = new Date();
+    const minuteOffset = startTimestamp.getUTCMinutes() - 5;
+
+    startTimestamp.setUTCMilliseconds(0);
+    startTimestamp.setUTCSeconds(0);
+    startTimestamp.setUTCMinutes(minuteOffset);
+
+    const { data: alerts, error: alertsError } = await supabaseClient
+        .from<AlertRecord>(TABLES.ALERT_HISTORY)
+        .select("*")
+        .eq("alert_type", dataProcessingAlertType)
+        .is("resolved_at", null)
+        .gt("fired_at", startTimestamp.toUTCString());
+
+    if (alertsError !== null) {
+        returnPostgresError(alertsError);
+    }
+
+    const { data: confirmations, error: confirmationsError } = await supabaseClient
+        .from<AlertRecord>(TABLES.ALERT_HISTORY)
+        .select("*")
+        .eq("alert_type", dataProcessingAlertType)
+        .gt("resolved_at", startTimestamp.toUTCString());
+
+    if (confirmationsError !== null) {
+        returnPostgresError(confirmationsError);
+    }
+
+    if (isEmpty(alerts) && isEmpty(confirmations)) {
+        // Terminate the function without error if there aren't any active notifications in the system.
+        return new Response(null, {
+            headers: { ...corsHeaders, "Content-Type": "application/json" },
+            status: 200,
+        });
+    }
+
+    const pendingAlertEmails: EmailConfig[] = alerts
+        ? alerts.map(
+            ({
+                arguments: { emails, evaluation_interval, spec_type },
+                catalog_name,
+            }) => {
+                let formattedEvaluationInterval = evaluation_interval;
+
+                // A postgresql interval in hour increments has the following format: 'HH:00:00'.
+                if (evaluation_interval.includes(":")) {
+                    const timeOffset = evaluation_interval.split(":");
+                    const hours = Number(timeOffset[0]);
+
+                    formattedEvaluationInterval = isFinite(hours) ? `${hours} hours` : `${timeOffset[0]} hours`;
```

Comment:
`timeOffset` corresponds to the hour substring of the evaluation interval, taking the format `HH` (e.g., `'02'`, `'04'`, `'12'`). Ideally, an hour-based interval that is less than ten would be represented by a single digit (e.g., a two-hour evaluation interval, `'02:00:00'`, should be translated to the following substring in the email: `2 hours`). To accomplish this, the hour segment of the evaluation interval is selected (i.e., `timeOffset[0]`) and that substring is attempted to be converted to a number. This conditional is a failsafe, in the event the aforementioned conversion fails which would result in the display of two digits for the hour (e.g., `02 hours` instead of `2 hours`).

Reply:
Makes sense, thanks for the explanation. Might be good to leave a comment as to the purpose, in case others get confused by that too.